src/Microsoft.Management.Deployment/CatalogPackage.cpp

```diff
@@ -126,15 +126,27 @@ namespace winrt::Microsoft::Management::Deployment::implementation
                     auto latestVersionImpl = winrt::make_self<wil::details::module_count_wrapper<
                         winrt::Microsoft::Management::Deployment::implementation::PackageVersionInfo>>();
                     latestVersionImpl->Initialize(std::move(latestApplicableVersion));
-                    m_defaultInstallVersion = *latestVersionImpl;
+                    m_latestApplicableVersion = *latestVersionImpl;
                 }
             });
     }
 
     Microsoft::Management::Deployment::PackageVersionInfo CatalogPackage::DefaultInstallVersion()
     {
-        InitializeDefaultInstallVersion();
-        return m_defaultInstallVersion;
+        InitializeLatestApplicableVersion();
+
+        if (m_latestApplicableVersion)

```

Comment:
If people absolutely do not want applicability check for DefaultInstallVersion, I can remove this if statement. But I feel having applicability check will lower chances of upgrade failures.